### ISSUE TITLE
[1.0.0.alpha.19] Linear-gradients no longer converted to SVG

### ISSUE THREAD

**Issue by thany** [Compass/compass#1676]
`linear-gradient` and `repeating-linear-gradient` are no longer converted to SVG for old Android, Opera and IE9. When I enable `$debug-browser-support`, it does output a comment that says the `-ms-` prefix wasn't included, even though `-ms-linear-gradient` doesn't even exist?? Regardless, it says nothing about why the SVG version wasn't outputted, even though IE9 is supposed to be supported.

Is this a bug? Or how can I force Compass to output SVG gradients for IE9, or force IE9 to be supported?

Mind you, this does not help:

```
$browser-minimum-versions: ('ie': '9');
```

But what will?

**Comment by chriseppstein** [comment]
I added a test case in 95cb9eb8eb665790831b04666151dd438c510959 but I cannot reproduce the issue.

There is special case handling for this here: https://github.com/chriseppstein/compass/blob/master/core/stylesheets/compass/css3/_images.scss#L32

Maybe this is just some other config issue in your project?

**Comment by chriseppstein** [comment]
Please provide a failing test case as a simple compass project if you think I'm mistaken.

**Comment by thany** [comment]
No problem, here we are:
http://sassmeister.com/gist/970ffe312a2d6dcdb5e5

This is fairly accurate, as initially I didn't have any browser config defined, and the output is similar to what I'm getting (i.e. no SVG).

**Comment by Arcovion** [comment]
I'm also getting this, please reopen.

It should work by default; [IE9 needs it](http://caniuse.com/#feat=css-gradients) - but as far as I can tell it can't be enabled at all. =[

**Comment by chriseppstein** [comment]
ok. First, you need to set the threshold lower. Then it turns out that the issue is that it only works with the new "to position" style gradients:

http://sassmeister.com/gist/74a94d6a6f04fea03f16

Degrees don't get a svg gradient for some reason.

**Comment by Arcovion** [comment]
Cool, so `linear-gradient(left, *)` works, it just doesn't show up without `$browser-minimum-versions: (ie: '9')`.

I'm new to the threshold system but I don't really understand it. Why is IE9 omitted when the `$critical-usage-threshold` is so low? I thought it used [this data](http://caniuse.com/usage_table.php) but it's clear that wouldn't work because of IE6 etc... I guess you are manually removing IE9?

Anyway, how can I get a list of all browsers and supported versions inside Sass so I can see how they change when varying the `$critical-usage-threshold` and other options? I'd like to hard-code the necessary `$browser-minimum-versions` now; I guess I can set the threshold to 100% and rely solely on `$browser-minimum-versions` if needed.

**Comment by mirisuzanne** [comment]
linear-gradients don't use the `$critical-usage-threshold` by default. They use `$graceful-usage-threshold`:

``` scss
$gradient-support-threshold: $graceful-usage-threshold !default;
```

**Comment by Arcovion** [comment]
I see, still it shows `Content for ie 9 omitted` with any value though. In my mind setting both thresholds to 0 would enable _all_ browsers and 100 would disable all of them, the latter works fine.
`$svg-gradient-shim-threshold` is not connected to the global thresholds.

**Comment by thany** [comment]
> Degrees don't get a svg gradient for some reason.

That's a bug that will be fixed, right? Because afaik, the degrees (or radians) notation is still accepted in the standard. And in this particular case, I really did need `45deg`, a `to bottom right` wouldn't look like what I need.

Apart from that, I do have to say, perhaps there are a few too many variables to control browser prefixes/workarounds. I think if there will be as many as there are now, issues will be coming in with folks who don't understand why some is or isn't outputted, especailly when 1.0 goes final... Just saying :)

**Comment by chriseppstein** [comment]
> That's a bug that will be fixed, right?

Right.

> Apart from that, I do have to say, perhaps there are a few too many variables to control browser prefixes/workarounds.

Agree. I want it to work intuitively but also to be configurable enough for users who have special needs. I think we can make `$svg-gradient-shim-threshold: $graceful-usage-threshold !default;` the standard. If people want to avoid the bloat, they can adjust that threshold up.

**Comment by thany** [comment]
Additionally, `$browser-minimum-versions: ('ie': '9');` should still force enable IE9 workarounds regardless of the threshold(s), right?

**Comment by Arcovion** [comment]
Note this comment is still true on `master`: https://github.com/chriseppstein/compass/issues/965#issuecomment-29696367
The new syntax `to right` throws a warning: `Cannot determine the opposite position of: to`

**Comment by chriseppstein** [comment]
Fixed by 1f20e44f4d36e37ffc4244615a138b876f1368bb
